From 66af888044d71c81a5568122efbddeb568da79d1 Mon Sep 17 00:00:00 2001
From: Franck LENORMAND <franck.lenormand@nxp.com>
Date: Fri, 19 Jan 2018 15:52:42 +0100
Subject: [PATCH 3236/5242] MLK-17412-02: Add support of secvio driver for
 imx8mq

commit  e112a9e93258143435c5716f61f06e510369bb2d from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch:
 - Adds compilation of the secvio driver for all arm64 targets
 - Adds the secvio driver to fsl-imx8mq.dtsi

Signed-off-by: Franck LENORMAND <franck.lenormand@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 4c4cc4a4..b0cb694 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -1307,6 +1307,20 @@
 		};
 	};
 
+	caam_snvs: caam-snvs@30370000 {
+		compatible = "fsl,imx6q-caam-snvs";
+		reg = <0 0x30370000 0 0x10000>;
+	};
+
+	irq_sec_vio: caam_secvio {
+		compatible = "fsl,imx7d-caam-secvio", "fsl,imx6q-caam-secvio";
+		interrupts = <GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
+		jtag-tamper = "disabled";
+		watchdog-tamper = "enabled";
+		internal-boot-tamper = "enabled";
+		external-pin-tamper = "disabled";
+	};
+
 	dma_apbh: dma-apbh@33000000 {
 		compatible = "fsl,imx7d-dma-apbh", "fsl,imx28-dma-apbh";
 		reg = <0 0x33000000 0 0x2000>;
-- 
1.7.9.5

